### ISSUE TITLE
Language detection still operating on paste despite `"workbench.editor.languageDetection": false`

### ISSUE THREAD

**Issue by OliverJAsh** [microsoft/vscode#146955]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.66.0
- OS Version: macOS 12.3.1

Steps to Reproduce:

1. `"workbench.editor.languageDetection": false`
2. Open a JSON file
3. Copy part of the file
4. Open a new untitled file
5. Paste

Expected: language is not detected as per setting

Actual: language is not detected as per setting

https://user-images.githubusercontent.com/921609/162072531-a5cadb3b-12ab-4337-b5c2-f6600c30bc2c.mov

**Comment by bpasero** [comment]
Yes as designed, copy from our editor preserves the source language mode.
